pkg/pod_lister/resolve_container.go

```diff
@@ -57,8 +57,14 @@ var (
 	cGroupIDToPath             = map[uint64]string{}
 
 	//regex to extract container ID from path
-	regexFindContainerIDPath          = regexp.MustCompile(`.*-(.*?)\.scope`)
-	regexReplaceContainerIDPathPrefix = regexp.MustCompile(`.*-`)
+	// regexFindContainerIDPath          = regexp.MustCompile(`.*-(.*?)\.scope`)
+
+	// regexFindContainerIDPath for cgroup v1
+	regexFindContainerIDPath          = regexp.MustCompile(`[0-9]*:.+slice.+`)
```

Comment:
make this path regex conditional based on the cgroup versions.

e.g. you can export the [isCGroupV2](https://github.com/sustainable-computing-io/kepler/blob/main/pkg/config/config.go#L76) and choose the right regex

Reply:
Added a method to check cgroup version and choose correct regex.